src/houghtransform.jl

```diff
@@ -1,84 +1,100 @@
-    function findlocalmaxima(accumulator_matrix::Array{Integer,2},threshold::Integer)
-        validLines = Vector{CartesianIndex}(0)
+    function findlocalmaxima!{T<:Integer}(validLines::AbstractVector{CartesianIndex{2}}, accumulator_matrix::Array{Int,2}, threshold::T)
         for val in CartesianRange(size(accumulator_matrix))
-            if  accumulator_matrix[val] > threshold &&
-                accumulator_matrix[val] > accumulator_matrix[val[1],val[2] - 1] &&
+            if  accumulator_matrix[val] >  threshold                             &&
+                accumulator_matrix[val] >  accumulator_matrix[val[1],val[2] - 1] &&
                 accumulator_matrix[val] >= accumulator_matrix[val[1],val[2] + 1] &&
-                accumulator_matrix[val] > accumulator_matrix[val[1] - 1,val[2]] &&
+                accumulator_matrix[val] >  accumulator_matrix[val[1] - 1,val[2]] &&
                 accumulator_matrix[val] >= accumulator_matrix[val[1] + 1,val[2]]
                 push!(validLines,val)
             end
         end
-        validLines
     end
 
     ρ > 0 || error("Discrete step size must be positive")
 
-    height, width = size(img)
+    height, width = indices(img)
     ρinv = 1 / ρ
     numangle = length(θ)
-    numrho = round(Integer,(2(width + height) + 1)*ρinv)
+    numrho = round(Int,(2(last(width) + last(height)) + 1)*ρinv)
 
-    accumulator_matrix = zeros(Integer, numangle + 2, numrho + 2)
+    accumulator_matrix = zeros(Int, numangle + 2, numrho + 2)
 
     #Pre-Computed sines and cosines in tables
     sinθ, cosθ = sin.(θ).*ρinv, cos.(θ).*ρinv
 
     #Hough Transform implementation
-    constadd = round(Integer,(numrho -1)/2)
+    constadd = round(Int,(numrho -1)/2)
     for pix in CartesianRange(size(img))
         if img[pix]
             for i in 1:numangle
-                dist = round(Integer, pix[1] * sinθ[i] + pix[2] * cosθ[i])
+                dist = round(Int, pix[1] * sinθ[i] + pix[2] * cosθ[i])
                 dist += constadd
                 accumulator_matrix[i + 1, dist + 1] += 1
             end
         end
     end
 
     #Finding local maximum lines
-    validLines = findlocalmaxima(accumulator_matrix, threshold)
+    validLines = Vector{CartesianIndex{2}}(0)
+    findlocalmaxima!(validLines, accumulator_matrix, threshold)
 
     #Sorting by value in accumulator_matrix
-    sort!(validLines, by = (x)->accumulator_matrix[x], rev = true)
+    @noinline sort_by_votes(validLines, accumulator_matrix) = sort!(validLines, lt = (a,b)-> accumulator_matrix[a]>accumulator_matrix[b])
+    sort_by_votes(validLines, accumulator_matrix)
 
     linesMax = min(linesMax, length(validLines))
 
-    lines = Vector{Tuple{Number,Number}}(0)
+    lines = Vector{Tuple{Float64,Float64}}(0)
 
     #Getting lines with Maximum value in accumulator_matrix && size(lines) < linesMax
-    for l in 1:linesMax
-        lrho = ((validLines[l][2]-1) - (numrho - 1)*0.5)*ρ
+    for l in range(1,linesMax)
```

Comment:
Any reason for this change? I think `1:linesMax` is more explicit, if the two are equivalent.